akka-stream/src/main/scala/akka/stream/impl/io/TLSActor.scala

```diff
@@ -46,12 +45,12 @@ object TLSActor {
 /**
  * INTERNAL API.
  */
-class TLSActor(
-  settings:          ActorMaterializerSettings,
-  sslContext:        SSLContext,
-  externalSslConfig: Option[AkkaSSLConfig],
-  firstSession:      NegotiateNewSession, role: TLSRole, closing: TLSClosing,
-  hostInfo: Option[(String, Int)], tracing: Boolean)
+private[stream] class TLSActor(
+  settings:        ActorMaterializerSettings,
+  createSSLEngine: ActorSystem ⇒ SSLEngine, // ActorSystem is only needed to support the AkkaSSLConfig legacy, see #21753
```

Comment:
Hm, is there any public API that forces us here to not fix the legacy requirement?